packages/toggle-core/test/Read/ChainSegmentFactoryTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pheature\Test\Core\Toggle\Read;
+
+use Pheature\Core\Toggle\Exception\InvalidSegmentTypeGiven;
+use Pheature\Core\Toggle\Read\ChainSegmentFactory;
+use Pheature\Core\Toggle\Read\Segment;
+use Pheature\Core\Toggle\Read\SegmentFactory;
+use PHPUnit\Framework\TestCase;
+
+final class ChainSegmentFactoryTest extends TestCase
+{
+    private const SEGMENT_ID = 'some_segment_id';
+    private const SEGMENT_TYPE = 'some_segment_type';
+    private const PAYLOAD = [
+        'some' => 'data',
+    ];
+
+    public function testItShouldThrowAnExceptionWhenItCantCreateAToggleStrategyType(): void
+    {
+        $this->expectException(InvalidSegmentTypeGiven::class);
+        $segmentFactory = $this->createMock(SegmentFactory::class);
+        $chainSegmentFactory = new ChainSegmentFactory($segmentFactory);
+
+        $chainSegmentFactory->create(self::SEGMENT_ID, self::SEGMENT_TYPE, self::PAYLOAD);
+    }
+
+    public function testItShouldBeCreatedWithAtLeastOneSegmentFactoryInstance(): void
+    {
+        $segmentFactory = $this->createMock(SegmentFactory::class);
+        $segmentFactory->expects(static::once())
+            ->method('types')
+            ->willReturn([self::SEGMENT_TYPE]);
+        $segmentFactory->expects(static::once())
+            ->method('create')
+            ->with(self::SEGMENT_ID, self::SEGMENT_TYPE, self::PAYLOAD)
+            ->willReturn($this->createMock(Segment::class));
+        $chainSegmentFactory = new ChainSegmentFactory($segmentFactory);
+
+        $chainSegmentFactory->create(self::SEGMENT_ID, self::SEGMENT_TYPE, self::PAYLOAD);
```

Comment:
In order to add an extra assert, we can declare a variable assigning the mocked segment that should be the same as the returned by the `ChainSegmentFactory`:

```suggestion
        $expectedSegment = $this->createMock(Segment::class);
        $segmentFactory->expects(static::once())
            ->method('create')
            ->with(self::SEGMENT_ID, self::SEGMENT_TYPE, self::PAYLOAD)
            ->willReturn($expectedSegment);
        $chainSegmentFactory = new ChainSegmentFactory($segmentFactory);

        $actual = $chainSegmentFactory->create(self::SEGMENT_ID, self::SEGMENT_TYPE, self::PAYLOAD);

        self::assertSame($expectedSegment, $actual);
```

What do you think?

Reply:
Regarding this change we could add the same logic for the changes of this PR: https://github.com/pheature-flags/pheature-flags/pull/158

Reply:
Thanks @xserrat ;-D I will make both changes :rocket::rocket::rocket:

Reply:
done;-D